rs/boundary_node/rate_limits/canister/confidentiality_formatting.rs

```diff
@@ -0,0 +1,87 @@
+use mockall::automock;
+use std::marker::PhantomData;
+
+use crate::{
+    access_control::{AccessLevel, ResolveAccessLevel},
+    types::{OutputConfig, OutputRuleMetadata},
+};
+
+/// Trait for formatting confidential data based on access levels
+#[automock(type Input = OutputConfig;)]
+pub trait ConfidentialityFormatting<Input = OutputConfig> {
+    type Input: Clone;
+
+    fn format(&self, value: &Self::Input) -> Self::Input;
+}
+
+/// A generic confidentiality formatter for various data types
+pub struct ConfidentialityFormatter<A, T> {
+    access_resolver: A,
+    phantom: PhantomData<T>,
+}
+
+impl<A, T> ConfidentialityFormatter<A, T> {
+    pub fn new(access_resolver: A) -> Self {
+        Self {
+            access_resolver,
+            phantom: PhantomData,
+        }
+    }
+}
+
+impl<A: ResolveAccessLevel> ConfidentialityFormatting
+    for ConfidentialityFormatter<A, OutputConfig>
+{
+    type Input = OutputConfig;
+
+    fn format(&self, config: &OutputConfig) -> OutputConfig {
+        let mut config = config.clone();
+        if self.access_resolver.get_access_level() == AccessLevel::RestrictedRead {
+            config.rules.iter_mut().for_each(|rule| {
+                if rule.disclosed_at.is_none() {
+                    rule.description = None;
+                    rule.rule_raw = None;
+                }
+            });
+        }
+        config
```

Comment:
```suggestion
        let mut config = config.clone();
        // return full config if authorized
        if self.access_resolver.get_access_level() == AccessLevel::FullAccess || self.access_resolver.get_access_level() == AccessLevel::FullRead {
            config
        }
        // return the redacted config otherwise
        config.rules.iter_mut().for_each(|rule| {
            if rule.disclosed_at.is_none() {
                rule.description = None;
                rule.rule_raw = None;
            }
        });
        config
```